python/tests/integration/thrift/test_tornado_client.py

```diff
@@ -71,37 +70,34 @@ def test_call(tchannel_server, service):
 
 
 @pytest.mark.gen_test
-@pytest.mark.xfail
 def test_protocol_error(tchannel_server, service):
-    tchannel_server.expect_call('Service::getItem').and_return(
-        messages.ErrorMessage(
-            code=messages.ErrorCode.bad_request,
-            message="stahp pls",
-        ),
-    )
+    tchannel_server.expect_call('wrong_endpoint')
 
     client = mk_client(service, tchannel_server.port)
     with pytest.raises(Thrift.TApplicationException) as excinfo:
         yield client.getItem("foo")
-    assert 'stahp' in str(excinfo.value)
+
+    print excinfo
+    assert (str(excinfo.value) == ("Endpoint 'Service::getItem' for" +
```

Comment:
You're testing the wrong aspect of this behavior. We want to test that an error frame we sent over the wire is translated into the correct exception. Can't you fix the server manager to support streams for this?

(Also, remove the print statement.)